5.1.no_vinaigrette.py

```diff
@@ -0,0 +1,53 @@
+"""
+Writer: Ranel Ben Simman Tov, 315967703
+"""
+
+import datetime as dt
+import random
+from enum import IntEnum
+
+
+class Weekday(IntEnum):
+    """
+    Enum for weekdays
+    """
+    MONDAY = 0
+
+
+def get_date(prompt_msg):
+    """
+    Get a date from the user
+    :param prompt_msg:
+    :return: date in YYYY-MM-DD format
+    """
+    try:
+        date = input(prompt_msg)
+        date = dt.date(*map(int, date.split('-')))  # create date object from input
+        return date
+    except (ValueError, TypeError):
+        print("Incorrect date format, should be YYYY-MM-DD")
+        return get_date(prompt_msg)
+
+
+def generate_random_date(date1, date2):
+    """
+    Generate a random date between date1 and date2
+    :param date1: the first date
+    :param date2: the second date
+    :return: random date between date1 and date2
+    """
+    return date1 + (date2 - date1) * random.random()
```

Comment:
You should not aim to write everything in a single line, even if you can.
It's better to save date2-date1 to a variable to make it more explicit and clear. 
This link https://towardsdatascience.com/you-should-stop-craving-for-python-one-liners-5dbf8e0de23d discuss this topic

Reply:
fixed